sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ReadFromKafkaDoFn.java

```diff
@@ -461,6 +461,11 @@ public ProcessContinuation processElement(
           return ProcessContinuation.resume();
         }
         for (ConsumerRecord<byte[], byte[]> rawRecord : rawRecords) {
+          // If the Kafka consumer returns a record with an offset that is already processed
```

Comment:
Maybe comment that the seek above may not be entirely successful and thus this is needed?

Reply:
Actually perhaps it is better to just check the position directly after the seek.  That would ensure this logic just covers the bad seek behavior and doesn't affect other processing (which could mask some other serious bug if records are arriving out of order).

We can query the consumer position after performign the seek to make sure it is correct.
https://kafka.apache.org/25/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html#position-org.apache.kafka.common.TopicPartition-java.time.Duration-